examples/chart/teleport-cluster/templates/config.yaml

```diff
@@ -60,7 +60,9 @@ data:
   {{- end }}
     kubernetes_service:
       enabled: true
-  {{- if not .Values.proxyListenerMode }}
+  {{- if eq .Values.proxyListenerMode "multiplex" }}
+      listen_addr: 0.0.0.0:443
+  {{- else if not .Values.proxyListenerMode }}
```

Comment:
To be honest I'm not quite sure I fully understand this change. Proxy listener mode only affects the Proxy endpoint so I'm not sure how it's related to the Kubernetes agent listen address.

I would think that the proper fix would be to remove the `if eq .Values.proxyListenerMode "multiplex"` condition and just always set listen_addr on the Kubernetes service? @smallinsky Could you confirm (or deny :))?

Reply:
IIRC if you're using multiplex mode and you don't explicitly specify a `listen_addr` for the Kubernetes service then it just doesn't enable the listener, which means that the service fails to start.

I think if anything the proper fix would be to make Teleport automatically set all `listen_addr`s correctly in multiplex mode. I've also noticed that the SSH listener doesn't start if you're in multiplex mode and you don't set `proxy_service.listen_addr` for example.

Reply:
AFIK When the `kubernetes_service` is running with auth and proxy services  the `kubernetes_service.listen_addr` needs to be provided no matter if proxy is using `multiplex` or `separate` mode. So  I think that this field should be `.Values.proxyListenerMode` in-depended.


When the  `kubernetes_service` is running in agent mode and connects to auth directly or through teleport proxy the `kubernetes_service.listen_addr` needs to empty.

The `proxy_service.listen_addr`  is used for local k8 dialling  and never exposed publicly.

Reply:
Got it. This is the `kubernetes_service` on the root `teleport-cluster` itself, so will need to always have a `listen_addr` set to host connections for the 'local' k8s cluster where the chart is running.

I think the correct change here is to remove the check for `.Values.proxyListenerMode` and just always set the `listen_addr` to `0.0.0.0:3027` regardless. As the listener is internal nothing else will change.

Maybe the fix needed is actually to change `proxy_service.kube_public_addr` to `:443` rather than `:3026` when multiplex mode is used?

Reply:
Yes, the `kubernetes_service.listen_addr` should be always set because there is no way to disable proxy_service in this yaml file so `kubernetes_service` will always run with `auth` and `proxy` services. This should solve the issue mentioned in this PR : 
> User Message: set kubernetes_service.listen_addr if this process can be reached from a teleport proxy or point teleport.auth_servers to a proxy to dial out

Regarding: 
> Maybe the fix needed is actually to change proxy_service.kube_listen_addr to 0.0.0.0:443 rather than 0.0.0.0:3026 when multiplex mode is used?

I think that current `proxy_service.kube_listen_addr` version is correct : 
```
  {{- if not .Values.proxyListenerMode }}
      kube_listen_addr: 0.0.0.0:3026
      mysql_listen_addr: 0.0.0.0:3036
  {{- end }}
  ```
changing `proxy_service.kube_listen_addr` will require  also to align k8[ service ](https://github.com/somethingnew2-0/teleport/blob/fix-kubernetes-multiplex/examples/chart/teleport-cluster/templates/service.yaml#L35 )

Reply:
I'm a little confused, what's the suggested solution?

Reply:
Removing the ` {{- if eq .Values.proxyListenerMode "multiplex" }}` IF condition from`kubernetes_service.listen_addr` logic so the `kubernetes_service.listen_addr` will be always set because `kubernetes_service.listen_addr` is not related to `proxyListenerMode` and same logic should by applied for multiplex and separate listener modes.

Reply:
Ah ok, simple enough. Done